src/Donations.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { Component } from 'react';
+import { formatCurrency, getDonations, Donation, track } from './utils';
+
+type State = {
+  loading: boolean;
+  donations: Donation[];
+  totalAmount: number;
+};
+
+class Donations extends Component {
+  state: State = {
+    loading: true,
+    donations: [],
+    totalAmount: 0,
+  };
+
+  async componentDidMount() {
+    const donations = this.state.donations;
+
+    const result = await getDonations();
+
+    for (var i = 0; i < result.length; i++) {
+      donations.push(result[i]);
+    }
+
+    this.setState({
+      donations: donations,
+      loading: false,
+      totalAmount: this.state.donations.reduce((p, c) => p + c.amount, 0),
+    });
+  }
+
+  renderDonation(donation: Donation) {
+    return (
+      <div>
+        <b style={{ textDecoration: 'underline' }}>
+          Donation amount:{' '}
+          {formatCurrency(donation.amount, donation.currencyCode)}
+        </b>
+        <span onClick={() => this.goToUrl(donation)}>Receipt</span>
+      </div>
+    );
+  }
+
+  goToUrl(donation: Donation) {
+    window.location.href = donation.receiptUrl;
+    track('receipt-click');
+  }
```

Comment:
See previous comment about just using an `<a>` tag... but also:
* only needs the `receiptUrl` string to be passed, not the entire `Donation` object
* firing `track` after re-directing - so function may never execute - should be the other way around